twitter/twitter.php

```diff
@@ -1138,27 +1138,34 @@ function twitter_expand_entities(App $a, $body, $item, $picture)
 	$tags_arr = [];
 
 	foreach ($item->entities->hashtags AS $hashtag) {
-		$url = "#[url=" . $a->get_baseurl() . "/search?tag=" . rawurlencode($hashtag->text) . "]" . $hashtag->text . "[/url]";
-		$tags_arr["#" . $hashtag->text] = $url;
-		$body = str_replace("#" . $hashtag->text, $url, $body);
+		$url = '#[url=' . $a->get_baseurl() . '/search?tag=' . rawurlencode($hashtag->text) . ']' . $hashtag->text . '[/url]';
+		$tags_arr['#' . $hashtag->text] = $url;
+		$body = str_replace('#' . $hashtag->text, $url, $body);
 	}
 
 	foreach ($item->entities->user_mentions AS $mention) {
-		$url = "@[url=https://twitter.com/" . rawurlencode($mention->screen_name) . "]" . $mention->screen_name . "[/url]";
-		$tags_arr["@" . $mention->screen_name] = $url;
-		$body = str_replace("@" . $mention->screen_name, $url, $body);
+		$url = '@[url=https://twitter.com/' . rawurlencode($mention->screen_name) . ']' . $mention->screen_name . '[/url]';
+		$tags_arr['@' . $mention->screen_name] = $url;
+		$body = str_replace('@' . $mention->screen_name, $url, $body);
 	}
 
 	if (isset($item->entities->urls)) {
-		$type = "";
-		$footerurl = "";
-		$footerlink = "";
-		$footer = "";
+		$type = '';
+		$footerurl = '';
+		$footerlink = '';
+		$footer = '';
 
 		foreach ($item->entities->urls as $url) {
 			$plain = str_replace($url->url, '', $plain);
 
 			if ($url->url && $url->expanded_url && $url->display_url) {
+				// Quote tweet, we just remove the quoted tweet URL from the body, the share block will be added later.
+				if (isset($item->quoted_status_id_str)
+					&& substr($url->expanded_url, -strlen($item->quoted_status_id_str)) == $item->quoted_status_id_str ) {
```

Comment:
This had confused me a little bit. Possibly it would be better to add some explanation like: "Check if the expanded url directs to the quoted tweet"

Reply:
I looked online for a visibly straghtforward way of checking that a substring is at the end of a string, but the other examples were worse.

Reply:
At first I thought "WTF is he doing there?" :grin: At situation like this I normally add some comment, so that others don't have to guess so much.